File: com/webservice/usuario.js
function isPalindrome(str) {
    let left = 0;
    let right = str.length - 1;
    while (left < right) {
        if (str.charAt(left) !== str.charAt(right)) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}

// Example usage
const input = "racecar"; // Example input
console.log(`Is '${input}' a palindrome? ${isPalindrome(input)}`);

File: com/webservice/usuario.java
public class usuario {
    public static void main(String[] args) {
        String input = "racecar"; // Example input
        System.out.println("Is '" + input + "' a palindrome? " + isPalindrome(input));
    }

    public static boolean isPalindrome(String str) {
        int left = 0;
        int right = str.length() - 1;
        while (left < right) {
            if (str.charAt(left) != str.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}